src/kokkos/test/kokkos/testEigenGPU.cc

```diff
@@ -0,0 +1,426 @@
+  fillHitsAndHitsCov(hits, hits_ge);
+}
+
+template <int N>
+void testFit() {
+  constexpr double B = 0.0113921;
+
+  Kokkos::View<double*, KokkosExecSpace> d_hits("d_hits", Rfit::maxNumberOfTracks() * sizeof(Rfit::Matrix3xNd<N>));
+  Kokkos::View<float*, KokkosExecSpace> d_hits_ge("d_hits_ge", Rfit::maxNumberOfTracks() * sizeof(Rfit::Matrix6xNf<N>));
+  Kokkos::View<double*, KokkosExecSpace> d_fast_fit_results("d_fast_fit_results",
+                                                            Rfit::maxNumberOfTracks() * sizeof(Vector4d));
+  Kokkos::View<Rfit::line_fit*, KokkosExecSpace> d_line_fit_results("d_line_fit_results",
+                                                                    Rfit::maxNumberOfTracks() * sizeof(Rfit::line_fit));
+  Kokkos::View<Rfit::circle_fit*, KokkosExecSpace> d_circle_fit_results(
+      "d_circle_fit_results", Rfit::maxNumberOfTracks() * sizeof(Rfit::circle_fit));
+
+  Rfit::Matrix3xNd<N> hits;
+  Rfit::Matrix6xNf<N> hits_ge = MatrixXf::Zero(6, N);
+
+  double* fast_fit_resultsGPUret = new double[Rfit::maxNumberOfTracks() * sizeof(Vector4d)];
+  Rfit::circle_fit* circle_fit_resultsGPUret = new Rfit::circle_fit();
+  Rfit::line_fit* line_fit_resultsGPUret = new Rfit::line_fit();
+
+  fillHitsAndHitsCov(hits, hits_ge);
+
+  std::cout << "sizes " << N << ' ' << sizeof(hits) << ' ' << sizeof(hits_ge) << ' ' << sizeof(Vector4d) << ' '
+            << sizeof(Rfit::line_fit) << ' ' << sizeof(Rfit::circle_fit) << std::endl;
+
+  std::cout << "Generated hits:\n" << hits << std::endl;
+  std::cout << "Generated cov:\n" << hits_ge << std::endl;
+
+  // FAST_FIT_CPU
+  Vector4d fast_fit_results;
+#ifdef USE_BL
+  BrokenLine::BL_Fast_fit(hits, fast_fit_results);
+#else
+  Rfit::Fast_fit(hits, fast_fit_results);
+#endif
+  std::cout << "Fitted values (FastFit, [X0, Y0, R, tan(theta)]):\n" << fast_fit_results << std::endl;
+
+  // cudaMemset d_fast_fit_results & d_line_fit_results to 0
+  int tmpN = Rfit::maxNumberOfTracks() * sizeof(Vector4d);
+  Kokkos::parallel_for(
+      "init_fast_hit_res",
+      Kokkos::RangePolicy<KokkosExecSpace>(KokkosExecSpace(), 0, tmpN),
+      KOKKOS_LAMBDA(const int& i) {
+        unsigned char* ptr = reinterpret_cast<unsigned char*>(d_fast_fit_results.data());
+        ptr[i] = 0;
+      });
+  tmpN = Rfit::maxNumberOfTracks() * sizeof(Rfit::line_fit);
+  Kokkos::parallel_for(
+      "init_line_fit_res",
+      Kokkos::RangePolicy<KokkosExecSpace>(KokkosExecSpace(), 0, tmpN),
+      KOKKOS_LAMBDA(const int& i) {
+        unsigned char* ptr = reinterpret_cast<unsigned char*>(d_line_fit_results.data());
+        ptr[i] = 0;
+      });
+
+  // for timing purposes we fit 4096 tracks
+  constexpr uint32_t Ntracks = 4096;
+
+  team_policy policy = team_policy(KokkosExecSpace(), Ntracks / 64, 64);
+
+  Kokkos::parallel_for(
+      "kernelPrintSizes", policy, KOKKOS_LAMBDA(const member_type& teamMember) {
+        kernelPrintSizes<N>(d_hits, d_hits_ge, teamMember);
```

Comment:
Is the hierarchical parallelism actually used somewhere there (I missed if it is)? Or in other words, why thread teams instead of just `RangePolicy`?